src/stdlib_string_type.fypp

```diff
@@ -431,7 +431,7 @@ contains
         type(string_type), intent(in) :: string
         integer :: ich
 
-        ich = merge(iachar(string%raw), 0, allocated(string%raw))
+        ich = merge(iachar(string%raw(1:1)), 0, allocated(string%raw))
```

Comment:
Will this work for an empty string?

Reply:
Thank you for pointing this out.
Since "iachar" only accepts a character type of size 1 as an argument, I checked it in the if statement at "git commit 1a87696".